runtime/v2/manager.go

```diff
@@ -223,12 +228,29 @@ func (m *MM[T]) ExportGenesisForModules(
 			continue
 		}
 
-		res, err := moduleI.ExportGenesis(ctx)
-		if err != nil {
-			return nil, err
+		channels[moduleName] = make(chan genesisResult)
+		go func(moduleI ModuleI, ch chan genesisResult) {
```

Comment:
Here we call `RunWithCtx` inside `RunWithCtx` (from appmanager).
If we need it here for all modules, we possibly don't need it in appmanager.